modules/core/src/main/java/org/locationtech/jts/geom/impl/CoordinateArraySequence.java

```diff
@@ -161,19 +175,19 @@ public CoordinateArraySequence(CoordinateSequence coordSeq)
    * 
    * @param array array is modified in place as needed
    */
-  protected void enforceArrayConsistency(Coordinate[] array)
+  protected void checkArrayConsistency(Coordinate[] array)
   {
-     Coordinate sample = createCoordinate();
-     Class<?> type = sample.getClass();
+     Class<?> type = null;
      for( int i = 0; i < array.length; i++) {
        Coordinate coordinate = array[i];
        if( coordinate == null ) {
-         array[i] = createCoordinate();
+         continue;
+       }
+       if (type == null) {
+         type = coordinate.getClass();
        }
-       else if(!coordinate.getClass().equals(type)) {
-         Coordinate duplicate = createCoordinate();
-         duplicate.setCoordinate(coordinate);
-         array[i] = duplicate;         
+       else  if(!coordinate.getClass().equals(type)) {
```

Comment:
Do we care about `isAssignableFrom`? Or are we expecting exactly `equals`?

Reply:
not sure... just copied what was there.  I think best to require exact equality.